api/login/login.controller.js

```diff
@@ -37,23 +37,24 @@ module.exports = {
       code_challenge: code_challenge,
       code_challenge_method: 'S256',
       state: state,
-      // nonce: generatedNonce
     });
 
     //save auth state details..........
     saveAuthStateDetails(state, code_challenge, code_verifier, generatedNonce, authUrl);
 
-    req.session.nonce = generatedNonce;
+    //save private key
+    const private_key = getMyPrivateKey();
+    await saveSigninKey(state,private_key);
```

Comment:
We should generate and save the signing key only at the startup of the application, if there are no signing keys available. The idea is not to use new signing keys for each authentication request, but to use the same signing key across all authentication requests until the signing key is expired. So think the saving operation should be moved to `app.js`

Reply:
i fixed it sir